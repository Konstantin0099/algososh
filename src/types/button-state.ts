export interface ButtonState {
    [name: string]: {
      isDisabled: boolean;
      isLoading?: boolean;
    };
  }