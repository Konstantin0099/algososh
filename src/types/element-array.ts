import { ElementStates } from "./element-states";
export interface ElementArray {
    letter?: string,
    state: ElementStates
  }