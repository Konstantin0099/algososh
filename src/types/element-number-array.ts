import { ElementStates } from "./element-states";
export interface ElementNumberArray {
    number: number,
    state: ElementStates
  }