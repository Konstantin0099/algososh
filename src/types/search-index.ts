export interface SearchIndex {
    index?: number;
    value?: string
  }