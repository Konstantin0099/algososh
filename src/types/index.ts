import { ElementArray } from "./element-array"
import { ElementNumberArray } from "./element-number-array"
import { ElementStates } from "./element-states"
import { Direction } from "./direction"
import { SortingMethod } from "./sorting-method"
import { ButtonState } from "./button-state"

export { ElementStates, Direction, SortingMethod }
export type { ElementArray, ElementNumberArray, ButtonState }