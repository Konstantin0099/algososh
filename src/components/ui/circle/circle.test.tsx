import { Circle } from "./circle";
import renderer from "react-test-renderer";
import { ElementStates } from "../../../types";

describe("+++Circle+++", () => {
  it("Circle!!!!!with letter", () => {
    const circleSnapshot = renderer.create(<Circle letter="letter" />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!without letter", () => {
    const circleSnapshot = renderer.create(<Circle />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with head=ReactElement", () => {
    const circleSnapshot = renderer
      .create(<Circle head={<Circle />} />)
      .toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with head=string", () => {
    const circleSnapshot = renderer.create(<Circle head={"head"} />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!without head", () => {
    const circleSnapshot = renderer.create(<Circle head={null} />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with tail=ReactElement", () => {
    const circleSnapshot = renderer
      .create(<Circle tail={<Circle />} />)
      .toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with tail=string", () => {
    const circleSnapshot = renderer.create(<Circle tail={"tail"} />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!without tail", () => {
    const circleSnapshot = renderer.create(<Circle tail={null} />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with index", () => {
    const circleSnapshot = renderer.create(<Circle index={10} />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with isSmall===true", () => {
    const circleSnapshot = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with state===default", () => {
    const circleSnapshot = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with state===changing", () => {
    const circleSnapshot = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
  it("Circle!!!!!with state===modified", () => {
    const circleSnapshot = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(circleSnapshot).toMatchSnapshot();
  });
});
