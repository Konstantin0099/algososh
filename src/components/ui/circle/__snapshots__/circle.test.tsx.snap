// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`+++Circle+++ Circle!!!!!with head=ReactElement 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with head=string 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    10
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with isSmall===true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      letter
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with state===changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with state===default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with state===modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!with tail=ReactElement 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`+++Circle+++ Circle!!!!!with tail=string 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`+++Circle+++ Circle!!!!!without head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`+++Circle+++ Circle!!!!!without tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
